connection_manager.go

```diff
@@ -181,6 +181,14 @@ func (n *connectionManager) HandleMonitorTick(now time.Time, p, nb, out []byte)
 			continue
 		}
 
+		// Does the vpnIp point to this hostinfo or is it ancillary? If we have ancillary hostinfos then we need to
+		// decide if this should be the main hostinfo if we are seeing traffic on it
+		primary, _ := n.hostMap.QueryVpnIp(hostinfo.vpnIp)
+		mainHostInfo := true
+		if primary != nil && primary != hostinfo {
+			mainHostInfo = false
+		}
```

Comment:
Is it okay that we don't have a lock on hostMap here? (something else could add an entry to the hostMap for this vpnIP after we do the query)

Reply:
We have the lock in `QueryVpnIp` so we will be able to determine if this hostinfo, that is seeing traffic, was primary or not at that instant.

The only things that I think can race here are a `closeTunnel and a new handshake completing. In both cases there they will cause a primary swap, so maybe there will be some jitter in the primary for a few connection manager ticks?

In short I think it should be fine :tm: